central/complianceoperator/v2/checkresults/datastore/datastore_impl.go

```diff
@@ -52,3 +73,124 @@ func (d *datastoreImpl) SearchCheckResults(ctx context.Context, query *v1.Query)
 
 	return d.store.GetByQuery(ctx, query)
 }
+
+// CheckResultStats retrieves the scan results stats specified by query
+func (d *datastoreImpl) CheckResultStats(ctx context.Context, query *v1.Query) ([]*ResourceCountByResultByCluster, error) {
```

Comment:
I wrote a separate ticket ROX-20429 to write a benchmark for this.  I'm curious how it will perform under a load so I think that benchmark will be valuable.